libraries/support/azure/vault.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Azure
+  class Vault
+    include Service
+
+    def initialize
+      @required_attrs = %i(rest_client)
+      @page_link_name = 'nextLink'
+    end
+
+    def keys
+      get(
+        url: '/keys',
+        api_version: '2016-10-01',
+        use_cache: false,
```

Comment:
Why don't we want to use the cache for keys and secrets?

Reply:
Caching had to be disabled in two places:
1. Because of how caching is performed in https://github.com/inspec/train/blob/master/lib/train/transports/azure.rb I found that requesting secrets from both `https://red.vault.azure.net` and `https://blue.vault.azure.net` would always return the secrets cached from whichever called first. Instead, we disable caching via `    inspec.backend.disable_cache(:api_call)` when we ask for a Vault client and always instantiate a new one, using the correct vault name.
2. The same problem was present in https://github.com/inspec/inspec-azure/blob/Add-Keys/libraries/support/azure/service.rb#L75 - if you request the `/secrets` or `/keys` endpoint more than once, even from different Vault clients, the result of the first call would always be returned.

Reply:
Sounds like a good work around at the moment. We should drop an issue to revisit caching later.